app/controllers/surveys_controller.rb

```diff
@@ -1,4 +1,58 @@
 class SurveysController < ApplicationController
   def send_survey
   end
+
+  def create
+    @survey = Survey.create!(survey_params)
+    questions = []
+
+    params[:questions].each do |question_params|
```

Comment:
What if you used `map` to build the questions array? Then you wouldn't need to push question on each iteration. I just get a cringing feeling everytime I see push on to a dynamic array


i.e.
```
questions = params[:questions].map do |question|
  question_params[:options].each do |response_params|
    ResponseChoice.Create...
  question = Question.create...
```